swift-sdk/Internal/IterableAPIInternal.swift

```diff
@@ -143,59 +103,94 @@ import UserNotifications
         register(token: token, appName: appName, pushServicePlatform: config.pushPlatform, onSuccess: onSuccess, onFailure: onFailure)
     }
 
-    /**
-     Disable this device's token in Iterable, for the current user.
-     */
-    @objc public func disableDeviceForCurrentUser() {
+    private func register(token: Data, appName: String, pushServicePlatform: PushServicePlatform, onSuccess: OnSuccessHandler?, onFailure: OnFailureHandler?) {
+        guard email != nil || userId != nil else {
+            ITBError("Both email and userId are nil")
+            onFailure?("Both email and userId are nil", nil)
+            return
+        }
+        
+        hexToken = (token as NSData).iteHexadecimalString()
+        
+        let device = UIDevice.current
+        let psp = IterableAPIInternal.pushServicePlatformToString(pushServicePlatform)
+        
+        typealias deviceKeys = ITBConsts.Device
+        var dataFields: [String : Any] = [
+            deviceKeys.localizedModel: device.localizedModel,
+            deviceKeys.userInterfaceIdiom: IterableAPIInternal.userInterfaceIdiomEnumToString(device.userInterfaceIdiom),
+            deviceKeys.systemName: device.systemName,
+            deviceKeys.systemVersion: device.systemVersion,
+            deviceKeys.model: device.model
+        ]
+        if let identifierForVendor = device.identifierForVendor?.uuidString {
+            dataFields[deviceKeys.identifierForVendor] = identifierForVendor
+        }
+        if let deviceId = localStorage.deviceId {
+            dataFields[deviceKeys.deviceId] = deviceId
+        }
+        if let sdkVersion = localStorage.sdkVersion {
+            dataFields[deviceKeys.iterableSdkVersion] = sdkVersion
+        }
+        if let appPackageName = Bundle.main.appPackageName {
+            dataFields[deviceKeys.appPackageName] = appPackageName
+        }
+        if let appVersion = Bundle.main.appVersion {
+            dataFields[deviceKeys.appVersion] = appVersion
+        }
+        if let appBuild = Bundle.main.appBuild {
+            dataFields[deviceKeys.appBuild] = appBuild
+        }
+        
+        let deviceDictionary: [String : Any] = [
+            ITBL_KEY_TOKEN: hexToken!,
+            ITBL_KEY_PLATFORM: psp,
+            ITBL_KEY_APPLICATION_NAME: appName,
+            ITBL_KEY_DATA_FIELDS: dataFields
+        ]
+        
+        var args: [String : Any]
+        if let email = email {
+            args = [
+                ITBL_KEY_EMAIL: email,
+                ITBL_KEY_DEVICE: deviceDictionary
+            ]
+        } else if let userId = userId {
+            args = [
+                ITBL_KEY_USER_ID: userId,
+                ITBL_KEY_DEVICE: deviceDictionary
+            ]
+        } else {
+            ITBError("Either email or userId is required.")
+            assertionFailure("either email or userId should be set")
+            args = [
```

Comment:
Is setting the args necessary if you are asserting an error?

Reply:
Also should we not attempt to send a request when the email/userId is not set.

Reply:
Please note that code will never enter line 165 above because of the guard statement at the beginning of the function. But I agree that it is not very clear. I will change the code so that it is more clear.